xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -257,12 +263,81 @@ protected void handleLdsResponse(String versionInfo, List<Any> resources, String
     }
   }
 
+  @VisibleForTesting
+  @Nullable // Returns null if the filter is optional but not supported.
+  static StructOrError<FilterConfig> parseHttpFilter(
+      io.envoyproxy.envoy.extensions.filters.network.http_connection_manager.v3.HttpFilter
+          httpFilter) {
+    String filterName = httpFilter.getName();
+    boolean isOptional = httpFilter.getIsOptional();
+    if (!httpFilter.hasTypedConfig()) {
+      if (isOptional) {
+        return null;
+      } else {
+        return StructOrError.fromError(
+            "HttpFilter [" + filterName + "] is not optional and has no typed config");
+      }
+    }
+    return parseRawFilterConfig(filterName, httpFilter.getTypedConfig(), isOptional, false);
+  }
+
+  @Nullable // Returns null if the filter should be ignored.
+  private static StructOrError<FilterConfig> parseRawFilterConfig(
+      String filterName, Any anyConfig, Boolean isOptional, boolean isOverrideConfig) {
+    checkArgument(
+        isOptional != null || isOverrideConfig, "isOptional can't be null for top-level config");
+    String typeUrl = anyConfig.getTypeUrl();
+    if (isOverrideConfig) {
+      isOptional = false;
+      if (typeUrl.equals(TYPE_URL_FILTER_CONFIG)) {
+        io.envoyproxy.envoy.config.route.v3.FilterConfig filterConfig;
+        try {
+          filterConfig =
+              anyConfig.unpack(io.envoyproxy.envoy.config.route.v3.FilterConfig.class);
+        } catch (InvalidProtocolBufferException e) {
+          return StructOrError.fromError("Invalid proto: " + e);
```

Comment:
Include the filter name in the error message now that you are completely relying on this method to give you an error message.

Or you could let the caller decorate the error message with filter name (then you do not need to pass in the filter name).

Reply:
Included the filter name in the error message now